Jellyfin.Api/Controllers/PlaylistsController.cs

```diff
@@ -221,6 +221,11 @@ public class PlaylistsController : BaseJellyfinApiController
             return userPermission;
         }
 
+        if (playlist.OwnerUserId.Equals(callingUserId))
+        {
+            return new PlaylistUserPermissions(callingUserId, true);
+        }
```

Comment:
I would move this if-statement to before the `var userPermission = xxx` line. Although unlikely, if for some reason the playlist contains a share with CanEdit set to false for the owner it would basically lock the owner out of his own playlist.